### PR TITLE
Refactor health controller

### PR DESCRIPTION
**Pull request by AbdullahAli**
Refactor the health controller 

- [x] removed `checkESPCanBeReached` since it was not used
- [x] remove `co` from `Health Controller`
- [x] ping db and queue on api call (not on interval)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,13 +58,14 @@
     "co-mocha": "^1.2.2",
     "eslint": "^3.5.0",
     "husky": "^0.14.3",
-    "istanbul": "^0.4.3",
+    "istanbul": "^1.0.0-alpha.2",
```
Comment (AbdullahAli) [inline package.json]: Needed as `istanbul` did not recognise the `async/await` syntax 🔪

File: test/controllers/send.test.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ const invalidDomainData = require('../fixtures/invalidFromDomainPostData');
 const uuidMalformedData = { thisIs: 'Wrong' };
 
 const sendByUuidUrl = '/send';
-const sandbox = sinon.sandbox.create();
+const sandbox = sinon.createSandbox();
```
Comment (AbdullahAli) [inline test/controllers/send.test.js]: Fix deprecation errors

File: app/controllers/health.controller.js [hunk 3]
```diff
@@ -1,164 +1,80 @@
...
+											'Wait a few minutes to see if the reboot has fixed the issue. If not, go back to resources tab on the Heroku console, ' +
+											'set all the dynos to 0, and reset them after a few seconds to their original value.',
+					lastUpdated: new Date().toISOString(),
+					ok: queueStatusResponse.up
+				}
+			]
+		});
+	} catch (err) {
+		// eslint-disable-next-line no-console
+		console.log(err);
```
Comment (AbdullahAli) [inline app/controllers/health.controller.js]: Should we log this somewhere?

**Comment by melanierogan** [comment]
Looking good 👍

File: app/routes/health.route.js [hunk 4]
```diff
@@ -1,12 +1,8 @@
-const coExpress = require('co-express');
```
Comment (AbdullahAli) [inline app/routes/health.route.js]: bye felicia

1ryhtb

File: app/controllers/health.controller.js [hunk 5]
```diff
@@ -1,164 +1,80 @@
...
+		.then((res) => ({ up: res.ok }))
+		.catch(() => ({ up: false }));
 }
 
-function runESPCheck() {
-	// defaults to true checkESPCanBeReached();
-	setInterval(checkESPCanBeReached, 15000);
+function dbStatus() {
+	return mongoose.connection.db.admin().ping()
+		.then((res) => ({ up: !!res.ok }))
```
Comment (keirog) [inline app/controllers/health.controller.js]: 😍

File: package.json [hunk 6]
```diff
@@ -58,13 +58,14 @@
...
     "mocha": "^2.5.3",
     "nock": "^9.4.4",
     "nodemon": "^1.18.3",
     "rewire": "^2.4.0",
-    "sinon": "^1.17.2",
+    "sinon": "^6.1.4",
```
Comment (keirog) [inline package.json]: 😮

Comment (AbdullahAli) [reply package.json]: 😎

File: app/controllers/health.controller.js (hunk 3 above)
Comment (keirog) [reply app/controllers/health.controller.js]: Ooh yep, if we use [our logger util](https://github.com/Financial-Times/email-send-api/blob/master/logger.js) it’ll always go to Splunk if you call `logger.error`. Best wrap it in another try/catch though (I know, bleurgh)—we want health checks to be as bullet proof as possible.

File: test/routes/health.route.test.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+			testUtils
+				.get('/__gtg')
+				.expect(() => expect(gtgSpy.calledOnceWith(sandbox.match(ServerResponse))))
+				.end(done);
+		});
+	});
+
+	describe('GET /__health', () => {
+		it('is handled by the HealthController health handler', (done) => {
```
[the remark this answers is not in the document]
Comment (AbdullahAli) [reply test/routes/health.route.test.js]: Fixed: https://github.com/Financial-Times/email-send-api/pull/258/commits/1841a7749162bacac8a3f92ef9d0f09812f9c0be 👍

File: app/controllers/health.controller.js (hunk 3 above)
Comment (AbdullahAli) [reply app/controllers/health.controller.js]: Sure 👍 

So wrap `logger.error` in a `try/catch`.  What should we do in the `catch` case? Can we fail silently?

Comment (keirog) [reply app/controllers/health.controller.js]: Maybe still a console.log as a last resort

Comment (AbdullahAli) [reply app/controllers/health.controller.js]: Fixed: https://github.com/Financial-Times/email-send-api/pull/258/commits/9e5e1437e4e76fb59b56723035e8f935e347447e 👍

**Comment by AbdullahAli** [comment]
Worth noting: you are no longer required to hold a reference to the `sandbox` since `sinon` is now a (default) `sandbox`
 object: https://sinonjs.org/guides/migrating-to-5.0



giphy 1

**c://** @Financial-Times/email-platform-engineers

**Pull request merged**